src/utilities/merge.ts

```diff
@@ -0,0 +1,29 @@
+import {GeneralObject} from '../types';
+
+export default function merge(...objs: GeneralObject[]) {
+  const final = {};
+
+  for (const obj of objs) {
+    mergeRecursively(final, obj);
+  }
+
+  return final;
+}
+
+function mergeRecursively(objA: GeneralObject, objB: GeneralObject) {
+  for (const key in objB) {
+    if (!objB.hasOwnProperty(key)) {
+      continue;
+    } else if (isMergableValue(objB[key]) && isMergableValue(objA[key])) {
```

Comment:
`isMergeableValue` ? Sorry to pick on spelling, I just find it easier to understand.